### PR TITLE
Retry fetching follower global checkpoint when it fails.

### LINKED ISSUE #34016: [CCR] unfollow API should be resilient to task not existing
After setting up CCR I deleted one of the primary shards of the leader index and restart the single node cluster. That caused a stats exception which I guess caused the persistent task for that shard to fail to start.

```
[2018-09-24T22:54:43,423][INFO ][o.e.x.c.a.ShardFollowTasksExecutor] [Boaz-Mac.local] [follow_index_2][1] Started to track leader shard [lead_index][1]
[2018-09-24T22:54:43,444][WARN ][o.e.p.PersistentTasksNodeService] [Boaz-Mac.local] task f3k2vdkQTL-X_eRHvn_7VA-1 failed with an exception
java.lang.IllegalArgumentException: Cannot find shard stats for shard [follow_index_2][1]
	at org.elasticsearch.xpack.ccr.action.ShardFollowTasksExecutor.lambda$fetchGlobalCheckpoint$4(ShardFollowTasksExecutor.java:187) [x-pack-ccr-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
	at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:60) [elasticsearch-7.0.0-alpha1-SNAPSHOT.jar:7.0.0-alpha1-SNAPSHOT]
```

I tried to pause following on one of the following indices:

```
POST follow_index/_ccr/unfollow
```

but this caused the following error. It should be possible to pause, even if not all tasks are started.

```
{
  "error": {
    "root_cause": [
      {
        "type": "resource_not_found_exception",
        "reason": "the task with id Y2U0gXPBSaKJseUJEgqUCA-0 doesn't exist",
        "suppressed": [
          {
            "type": "resource_not_found_exception",
            "reason": "the task with id Y2U0gXPBSaKJseUJEgqUCA-1 doesn't exist"
          }
        ]
      }
    ],
    "type": "resource_not_found_exception",
    "reason": "the task with id Y2U0gXPBSaKJseUJEgqUCA-0 doesn't exist",
    "suppressed": [
      {
        "type": "resource_not_found_exception",
        "reason": "the task with id Y2U0gXPBSaKJseUJEgqUCA-1 doesn't exist"
      }
    ]
  },
  "status": 404
}
```

/cc @martijnvg @jasontedor

### PR DESCRIPTION
**Pull request by martijnvg**
Closes #34016

### CODE REVIEW

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java [hunk 1]
```diff
@@ -374,7 +374,7 @@ static long computeDelay(int currentRetry, long maxRetryDelayInMillis) {
         return Math.min(backOffDelay, maxRetryDelayInMillis);
     }
 
-    private static boolean shouldRetry(Exception e) {
+    static boolean shouldRetry(Exception e) {
```
Comment (martijnvg) [inline x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java]: Maybe `fetchGlobalCheckpoint()` should be done in `ShardFollowNodeTask` like we update mapping or fetch document from shard changes api then we don't need to make to method package protected and complete reuse the retry logic that exists in `ShardFollowNodeTask`?

**Comment by jasontedor** [comment]
@elasticmachine run gradle build tests

**Comment by jasontedor** [comment]
@elasticmachine run gradle build tests

**Comment by jasontedor** [comment]
@elasticmachine run gradle build tests

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java (hunk 1 above)
Comment (bleskes) [reply x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java]: On the fence on this one. I understand your reasons but I feel that ShardFollowNodeTask is complicated enough so we shouldn't move things in there unless strictly needed. As far as I can tell, the amount of code stays the same.

**Review by bleskes — APPROVED** [review]
Production code LGTM. Is there any way we can test this?

File: x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java (hunk 1 above)
Comment (martijnvg) [reply x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java]: I see, I'm still undecided. Lets leave it this way.

**Comment by martijnvg** [comment]
> Is there any way we can test this?

I think when `ShardChangesIT#testFollowIndexAndCloseNode()` is [unmuted](https://github.com/elastic/elasticsearch/issues/33337) this should trigger conditions that allow the fetching of global checkpoints to be retried.

**Pull request merged**